### ISSUE TITLE
xcresulttool error in Xcode 16 beta 3

### ISSUE THREAD

**Issue by VIadislavJevremovic** [codemagic-ci-cd/cli-tools#423]
As of Xcode 16 beta 3, xcresulttool was updated after which errors are encountered while collecting test results:

```
[12:08:47] ERROR > Parsing test results failed
Failed to get result bundle object from /Users/builder/Library/Developer/Xcode/DerivedData/#####.xcresult
Error: This command is deprecated and will be removed in a future release, --legacy flag is required to use it.
Usage: xcresulttool get object [--legacy] --path <path> [--id <id>] [--version <version>] [--format <format>]
  See 'xcresulttool get object --help' for more information.
```

fastlane had this as well (https://github.com/fastlane/fastlane/issues/22132, fix https://github.com/dokimyj/fastlane/pull/1/files) with the solution being to include `--legacy` if xcresulttool version is 23021 and higher.

**Comment by priitlatt** [comment]
@VIadislavJevremovic, thanks for reporting this! We'll look into it soon.

**Comment by priitlatt** [comment]
@VIadislavJevremovic fix for that was just released with [version 0.53.4](https://github.com/codemagic-ci-cd/cli-tools/releases/tag/v0.53.4).
